### PR TITLE
Stats: Deprecate the feature flag stats/new-main-chart

### PR DESCRIPTION
**Pull request by dognose24**
#### Proposed Changes

* Refactor a few modernized components to coexist with legacy styles.
* Deprecate the legacy HTML structure in pages and components.

#### Testing Instructions

> Note: The instructions to show the `Store` and `Ads` pages can be found at https://github.com/Automattic/wp-calypso/pull/69661 and https://github.com/Automattic/wp-calypso/pull/69743.

* Spin up this change with the Calypso Live link.
* Navigate to the Stats page `Traffic`, `Insights`, `Store`, and `Ads`.
* Ensure the styles are the same as the previous ones.

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to #70209

### CODE REVIEW

**Comment by kangzj** [comment]
Hi @dognose24 the PR is pretty big to review. Is it okay for you to provide some inline comments stating what the changes are specifically for? That would be super helpful. Thanks.

File: client/components/chart/style.scss [hunk 1]
```diff
@@ -1,15 +1,18 @@
+@import "@wordpress/base-styles/breakpoints";
+
+$barChartBorderRadius: 2px;
+$yearBarChartBorderRadius: 4px;
```
Comment (jsnmoon) [inline client/components/chart/style.scss:4]: The naming convention in stylesheets is dash-case, so these lines should be:

```suggestion
$bar-chart-border-radius: 2px;
$year-bar-chart-border-radius: 4px;
```

File: client/components/chart/style.scss [hunk 2]
```diff
@@ -187,49 +205,45 @@ $y-axis-padding: 0 20px 0 10px;
...
 	list-style-type: none;
 	margin: 0;
 	font-size: $font-body-extra-small;
-	text-transform: uppercase;
-	letter-spacing: 0.1em;
 
-	@include breakpoint-deprecated( "<480px" ) {
-		width: 100%;
+	@media ( max-width: $break-mobile ) {
+		// width: 100%;
```
Comment (jsnmoon) [inline client/components/chart/style.scss:245]:
```suggestion
```

File: client/my-sites/stats/controller.jsx [hunk 3]
```diff
@@ -189,7 +189,7 @@ export function overview( context, next ) {
 		return (
 			context.params.period === filter.period ||
 			context.path.indexOf( filter.path ) >= 0 ||
-			( filter.altPaths && context.path.indexOf( filter.altPaths ) >= 0 )
+			( filter.altPaths && filter.altPaths.includes( context.path ) )
```
Comment (jsnmoon) [inline client/my-sites/stats/controller.jsx:192]: Changes to the controller are unrelated to this PR, so let's omit them from here and propose them in a separate PR.

**Review by jsnmoon — CHANGES_REQUESTED** [review]
I agree with Jasper. Could we split this PR into multiple sub-PRs?

We could spin up three parallel PRs like this, for example:

1. PR for deprecating using the feature flag in the navigation-related components
2. PR for deprecating using the feature flag in the StatsSite component
3. PR for deprecating using the feature flag in the Ads and Store pages
- After merging the above PRs, a final PR for removing the feature flag from the JSON config files.

This breakdown is merely an example; I think you would have a better idea of how to split this PR into multiple parts.

**Comment by dognose24** [comment]
That sounds like a better plan. I will close this PR and open the separate PRs.

**Comment by dognose24** [comment]
@kangzj @jsnmoon Thanks for the feedback; which sounds like a better plan. I will close this PR and open the separate PRs.

**Pull request closed without merging**